src/multicellular_stage/editor/CellBodyPlanEditorComponent.cs

```diff
@@ -979,11 +1066,145 @@ private void ClearSelectedAction()
         EmitSignal(SignalName.OnCellTypeToEditSelected, default(Variant), false);
     }
 
+    private void OnEnergyBalanceOptionsChanged()
+    {
+        CalculateEnergyAndCompoundBalance(editedMicrobeCells);
+    }
+
+    private void OnResourceLimitingModeChanged()
+    {
+        CalculateEnergyAndCompoundBalance(editedMicrobeCells);
+    }
+
     private void OnCellsChanged()
     {
         UpdateAlreadyPlacedVisuals();
 
+        UpdateStats();
+
         UpdateArrow();
+
+        UpdateFinishButtonWarningVisibility();
+    }
+
+    private void UpdateStats()
+    {
+        organismStatisticsPanel.UpdateStorage(GetAdditionalCapacities(out var nominalCapacity), nominalCapacity);
+        organismStatisticsPanel.UpdateSpeed(CellBodyPlanInternalCalculations.CalculateSpeed(editedMicrobeCells));
+        organismStatisticsPanel.UpdateRotationSpeed(
+            CellBodyPlanInternalCalculations.CalculateRotationSpeed(editedMicrobeCells));
+
+        CalculateEnergyAndCompoundBalance(editedMicrobeCells);
+    }
+
+    /// <summary>
+    ///   Calculates the energy balance and compound balance for a colony
+    /// </summary>
+    private void CalculateEnergyAndCompoundBalance(IReadOnlyList<HexWithData<CellTemplate>> cells,
+        BiomeConditions? biome = null)
+    {
+        biome ??= Editor.CurrentPatch.Biome;
+
+        bool moving = organismStatisticsPanel.CalculateBalancesWhenMoving;
+
+        IBiomeConditions conditionsData = biome;
+
+        if (organismStatisticsPanel.ResourceLimitingMode != ResourceLimitingMode.AllResources)
+        {
+            conditionsData = new BiomeResourceLimiterAdapter(organismStatisticsPanel.ResourceLimitingMode,
+                conditionsData);
+        }
+
+        var energyBalance = new EnergyBalanceInfo();
+        energyBalance.SetupTrackingForRequiredCompounds();
+
+        // TODO: improve performance by calculating the balance per cell type
+        foreach (var hex in cells)
+        {
+            ProcessSystem.ComputeEnergyBalance(hex.Data!.Organelles, conditionsData, hex.Data.MembraneType, moving,
+                true, Editor.CurrentGame.GameWorld.WorldSettings, organismStatisticsPanel.CompoundAmountType,
+                energyBalance);
+        }
+
+        energyBalanceInfo = energyBalance;
+
+        organismStatisticsPanel.UpdateEnergyBalance(energyBalance);
+
+        float nominalStorage = 0;
+        Dictionary<Compound, float>? specificStorages = null;
```

Comment:
I had a bit of trouble understanding the ref usage here, so could you add a comment before these two variables that they are passed by ref so that they can be set by the first call to the method that uses them and then reused?